src/js/program/media-controller.js

```diff
@@ -316,14 +316,15 @@ export default class MediaController extends Events {
     set volume(volume) {
         this.provider.volume(volume);
     }
+
+    addProviderListeners() {
+        this.provider.off();
+        this.provider.on('all', this.providerListener, this);
+    }
 }
 
 function syncPlayerWithMediaModel(mediaModel) {
     // Sync player state with mediaModel state
     const mediaState = mediaModel.get('mediaState');
     mediaModel.trigger('change:mediaState', mediaModel, mediaState, mediaState);
 }
-
-function addProviderListeners(mediaController) {
-    mediaController.provider.on('all', mediaController.providerListener, mediaController);
```

Comment:
assuming `mediaController.provider` is not always the provider instantiated with _this_ media controller? or is this just to make things safer?

Reply:
yeah I think this was just a helper. I didn't need to add it to the interface and can move this code into the constructor. I only moved it because I considered calling it again.